tests/test_threadpool_limits.py

```diff
@@ -0,0 +1,189 @@
+import os
+import pytest
+
+
+from loky import cpu_count
+from loky.backend import thread_pool_limits
+from loky.backend._thread_pool_limiters import _CLibsWrapper
+from loky.backend._thread_pool_limiters import NOT_ACCESSIBLE
+from loky.backend._thread_pool_limiters import get_thread_limits
+from loky.backend._thread_pool_limiters import _set_thread_limits
+from loky.backend._thread_pool_limiters import openmp_num_threads
+
+from .utils import with_parallel_sum, with_numpy
+
+
+def should_skip_module(module):
+    return module['name'] == "openblas" and module['version'] == NOT_ACCESSIBLE
+
+
+def test_openmp_num_threads():
+    with pytest.raises(ValueError, message="n_threads=0 is invalid"):
+        openmp_num_threads(0)
+
+    assert openmp_num_threads(1) == 1
+    assert openmp_num_threads(2) == 2
+    assert openmp_num_threads(3) == 3
+
+    # Assert negative value give the right number
+    assert openmp_num_threads(-1) == cpu_count()
+    assert openmp_num_threads(-2) == max(1, cpu_count() - 1)
+    assert openmp_num_threads(-3) == max(1, cpu_count() - 2)
+
+
+@pytest.mark.parametrize("library", _CLibsWrapper.SUPPORTED_CLIBS)
+def test_thread_pool_limits(openblas_test_noskip, mkl_win32_test_noskip,
+                            library):
+    old_limits = get_thread_limits()
+    old_limits = {clib['name']: clib['n_thread'] for clib in old_limits}
```

Comment:
dict comprehensions are not supported in py2. `old_limits = dict((clib['name'], clib['n_thread']) for clib in old_limits)`

Reply:
Are you sure? It seems to pass the test in `python2.7` locally, on travis and appveyor.